rmc/utils/constraint.py

```diff
@@ -955,214 +955,199 @@ def get_rescue_transcripts_and_create_no_breaks_ht(overwrite: bool) -> None:
-    # Use `break_list` annotation (list of booleans for whether a row is a breakpoint)
-    # to filter one/additional breaks ht to rows that are significant breakpoints ONLY
-    ht = ht.filter(ht.break_list.any(lambda x: x))
-
-    # Group HT (filtered to breakpoint positions only) by transcript and
-    # count the number of breakpoints associated with each transcript
-    group_ht = ht.group_by("transcript").aggregate(n_breaks=hl.agg.count())
-
-    # Checkpoint to force hail to finish this group by computation
-    group_ht = group_ht.checkpoint(
-        f"{TEMP_PATH}/breaks_per_transcript.ht", overwrite=True
+    if len(round_nums) < 2:
+        raise DataException(
+            "At least two rounds of break search are needed if evidence of RMC is found, please double-check!"
+        )
+
+    logger.info(
+        "Collecting and merging no-break HTs from each search round starting at round 2..."
+    )
+    hts = []
+    for search_num in round_nums[1:]:
+        # For each search round:
+        # Get locus-level merged no-break table (no breaks in both single and simultaneous searches)
+        ht = merge_round_no_break_ht(is_rescue=False, search_num=search_num)
+        # Group to section-level and retain section obs- and exp-related annotations
+        ht = ht.group_by("section").aggregate(
+            section_obs=hl.agg.take(ht.section_obs, 1)[0],
+            section_exp=hl.agg.take(ht.section_exp, 1)[0],
+            section_oe=hl.agg.take(ht.section_oe, 1)[0],
+        )
+        hts.append(ht)
+    rmc_ht = hts[0].union(*hts[1:])
+    # Calculate chi-square value for each section
+    rmc_ht = rmc_ht.annotate(
+        section_chisq=calculate_section_chisq(rmc_ht.section_obs, rmc_ht.section_exp)
     )
+    # Annotate search pathway type (initial or rescue)
+    rmc_ht = rmc_ht.annotate(search_type="rescue" if is_rescue else "initial")
 
-    for i in range(1, max_n_breaks + 1):
-        temp_ht = group_ht.filter(group_ht.n_breaks == i)
-        transcripts = temp_ht.aggregate(hl.agg.collect_as_set(temp_ht.transcript))
-        if len(transcripts > 0):
-            transcripts_per_break[i] = transcripts
-        else:
-            transcripts_per_break[i] = hl.missing(hl.tset(hl.tstr))
-    return transcripts_per_break
+    # Convert section label to transcript and start and end positions
+    rmc_ht = rmc_ht.key_by()
+    rmc_ht = rmc_ht.transmute(
+        transcript=rmc_ht.section.split("_")[0],
+        start_pos=hl.int(rmc_ht.section.split("_")[1]),
+        end_pos=hl.int(rmc_ht.section.split("_")[2]),
+    )
+    # Convert start and end positions to interval
+    rmc_ht = rmc_ht.annotate(chr=gene_model.ht()[rmc_ht.transcript].chrom)
+    rmc_ht = rmc_ht.transmute(
+        interval=hl.parse_locus_interval(
+            hl.format(
+                "[%s:%s-%s]",
+                rmc_ht.chr,
+                rmc_ht.start_pos,
+                rmc_ht.end_pos,
+            )
+        ),
+    )
```

Comment:
```suggestion
    rmc_ht = rmc_ht.transmute(
        interval=hl.parse_locus_interval(
            hl.format(
                "[%s:%s-%s]",
                rmc_ht.locus.contig,
                rmc_ht.start_pos,
                rmc_ht.end_pos,
            )
        ),
    )
```
the HT read in with `single_search_round_ht_path` and `is_break_found=False` above still has a locus annotation, right? so this method would avoid an additional table join

Reply:
Addressed in new changes